packages/viz/src/lib/sources/DOSource.ts

```diff
@@ -79,28 +129,59 @@ export class DOSource extends Source {
     return parseFetchJSON(r);
   }
 
-  public async getLayerProps(): Promise<DataObservatoryLayerProps> {
-    const vizURL = `${this._baseURL}/visualization`;
-    const { apiKey } = this._credentials;
+  private async _getGeography(geographyID: string): Promise<Geography> {
+    const url = `${this._baseURL}/metadata/geographies/${geographyID}`;
+    const r = await fetch(url);
+    return parseFetchJSON(r);
+  }
 
+  public async init(): Promise<boolean> {
     // Get geography from metadata
-    const variable = await this._getVariable();
+    const variable = await this._getVariable(this._variable);
     const dataset = await this._getDataset(variable.dataset_id);
+    const geography = await this._getGeography(dataset.geography_id);
+
+    this._model = { dataset, variable, geography };
+    this.isInitialize = true;
+    return this.isInitialize;
+  }
 
-    const geography = dataset.geography_id;
+  public getLayerProps(): DOLayerProps {
+    if (!this.isInitialize || this._model === undefined)
+      throw new Error('To run this method you need to first call to init()');
+
+    const vizURL = `${this._baseURL}/visualization`;
+    const { apiKey } = this._credentials;
+
+    // Get geography from metadata
+    const geography = this._model.dataset.geography_id;
 
     const geographiesURLTemplate = `${vizURL}/geographies/${geography}/{z}/{x}/{y}.mvt?api_key=${apiKey}`;
     const dataURLTemplate = `${vizURL}/variables/{z}/{x}/{y}.json?variable=${this._variable}&api_key=${apiKey}`;
-    const geometryType = 'MultiPolygon';
 
-    return { geographiesURLTemplate, dataURLTemplate, geometryType };
+    return { type: 'TileLayer', geographiesURLTemplate, dataURLTemplate };
```

Comment:
Are we loosing `geometryType` here on purpose?

Reply:
I just created a specific method getGeometryType outside of layer props, it happens on styles where you only need the geometry type and not the tiles info

Reply:
I think geometryType could be part of the Metadata, having just one 'parent' object for all these pieces... (to consider in upcoming PRs, not in this one)